File: webapp-net/Ugc/Data/Comment.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Sdl.Web.Modules.Ugc.Data
{
    public class Comment
    {
        public long Id { get; set; }
        public int ItemPublicationId { get; set; }
        public int ItemId { get; set; }
        public int ItemType { get; set; }
        public CommentDate CreationDate { get; set; }
        public CommentDate LastModifiedDate { get; set; }
        public string Content { get; set; }
        public User User { get; set; }
        public List<Comment> Children { get; set; }

        [JsonIgnore]
        public int Rating { get; set; } = 0;
    }

    public class CommentDate
    {
        [JsonIgnore]
        public DateTime DateTime { get; set; }

        public int DayOfMonth { get; set; }
        public string DayOfWeek { get; set; }
        public int DayOfYear { get; set; }
        public int Hour { get; set; }
        public int Minute { get; set; }
        public string Month { get; set; }
        public int MonthValue { get; set; }
        public int Nano { get; set; }
        public int Second { get; set; }
        public int Year { get; set; }
    }
}

File: webapp-java/dxa-module-ugc/src/main/java/com/sdl/dxa/modules/ugc/data/Comment.java
package com.sdl.dxa.modules.ugc.data;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonRawValue;

import lombok.Data;
import org.joda.time.DateTime;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

/**
 * <p>Ugc comment</p>
 */

@Data
public class Comment {
    @JsonProperty("id")
    private long id;

    @JsonProperty("parentId")
    private long parentId;

    @JsonProperty("itemPublicationId")
    private int itemPublicationId;

    @JsonProperty("itemId")
    private int itemId;

    @JsonProperty("itemType")
    private int itemType;

    @JsonProperty(value = "creationDate")
    @JsonRawValue
    private String creationDate;

    @JsonProperty("lastModifiedDate")
    @JsonRawValue
    private String lastModifiedDate;

    @JsonIgnore
    private DateTime creationDateTime;

    @JsonIgnore
    private DateTime lastModifiedDateTime;

    @JsonProperty("content")
    private String content;

    @JsonProperty("user")
    private User user;

    @JsonProperty("children")
    private List<Comment> children;

    @JsonIgnore
    private int rating = 0;

    @JsonProperty(value = "metadata")
    private Map<String, String> metadata = new HashMap<>();

}
